# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/gtest/videoconduit_unittests.cpp
# Commit: b2134c0ea6e8
# Full Hash: b2134c0ea6e85ce2ffc27a18e7f072efa71607ef
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-12-18 00:51:03
# Regressor Bug: 1408256
# File Overlap Count: 1
# Description:
#   Bug 1408256: Fix bug where renegotiation would un-set a previously negotiated max-fs. r=dminor
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D55890
# ==============================================================================

diff -r bfbff4e80023 -r b2134c0ea6e8 media/webrtc/signaling/gtest/videoconduit_unittests.cpp
--- a/media/webrtc/signaling/gtest/videoconduit_unittests.cpp	Tue Dec 17 19:19:01 2019 +0000
+++ b/media/webrtc/signaling/gtest/videoconduit_unittests.cpp	Tue Dec 17 19:45:00 2019 +0000
@@ -516,7 +516,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(640, 360, 1);
@@ -661,7 +661,7 @@
 TEST_F(VideoConduitTest, TestOnSinkWantsChanged) {
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   wants.max_pixel_count = 256000;
   EncodingConstraints constraints;
@@ -673,7 +673,7 @@
   codecConfig.mEncodingConstraints.maxFs = 0;
   mVideoConduit->ConfigureSendMediaCodec(&codecConfig);
   mVideoConduit->StartTransmitting();
-  mVideoConduit->OnSinkWantsChanged(wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
   SendVideoFrame(1920, 1080, 1);
   EXPECT_LE(sink->mVideoFrame.width() * sink->mVideoFrame.height(), 256000);
   videoStreams = mCall->CreateEncoderStreams(sink->mVideoFrame.width(),
@@ -684,7 +684,7 @@
 
   codecConfig.mEncodingConstraints.maxFs = 500;
   mVideoConduit->ConfigureSendMediaCodec(&codecConfig);
-  mVideoConduit->OnSinkWantsChanged(wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
   SendVideoFrame(1920, 1080, 2);
   EXPECT_LE(sink->mVideoFrame.width() * sink->mVideoFrame.height(),
             500 * 16 * 16);
@@ -696,7 +696,7 @@
 
   codecConfig.mEncodingConstraints.maxFs = 1000;
   mVideoConduit->ConfigureSendMediaCodec(&codecConfig);
-  mVideoConduit->OnSinkWantsChanged(wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
   SendVideoFrame(1920, 1080, 3);
   EXPECT_LE(sink->mVideoFrame.width() * sink->mVideoFrame.height(),
             1000 * 16 * 16);
@@ -709,7 +709,7 @@
   wants.max_pixel_count = 64000;
   codecConfig.mEncodingConstraints.maxFs = 500;
   mVideoConduit->ConfigureSendMediaCodec(&codecConfig);
-  mVideoConduit->OnSinkWantsChanged(wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
   SendVideoFrame(1920, 1080, 4);
   EXPECT_LE(sink->mVideoFrame.width() * sink->mVideoFrame.height(), 64000);
   videoStreams = mCall->CreateEncoderStreams(sink->mVideoFrame.width(),
@@ -742,7 +742,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
   mVideoConduit->StartTransmitting();
 
   // This should crop to 16-alignment to help with scaling
@@ -796,7 +796,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   std::vector<webrtc::VideoStream> videoStreams;
@@ -1142,7 +1142,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(1280, 720, 1);
@@ -1238,7 +1238,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   SendVideoFrame(1280, 720, 1);
   ASSERT_EQ(sink->mVideoFrame.width(), 1280);
@@ -1298,7 +1298,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(1280, 720, 1);
@@ -1320,7 +1320,7 @@
   ASSERT_EQ(sink->mOnFrameCount, 3U);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeMaxFs) {
@@ -1337,7 +1337,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(1280, 720, 1);
@@ -1361,7 +1361,7 @@
   // maxFs should not force pixel count above what a sink has requested.
   // We set 3600 macroblocks (16x16 pixels), so we request 3500 here.
   wants.max_pixel_count = 3500 * 16 * 16;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   SendVideoFrame(1280, 720, 4);
   ASSERT_EQ(sink->mVideoFrame.width(), 960);
@@ -1382,7 +1382,7 @@
   ASSERT_EQ(sink->mOnFrameCount, 6U);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeMaxFsNegotiatedThenSinkWants) {
@@ -1399,7 +1399,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   unsigned int frame = 0;
   mVideoConduit->StartTransmitting();
@@ -1411,7 +1411,7 @@
   ASSERT_EQ(sink->mOnFrameCount, frame);
 
   wants.max_pixel_count = 3600 * 16 * 16;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   SendVideoFrame(1280, 720, frame++);
   ASSERT_EQ(sink->mVideoFrame.width(), 960);
@@ -1420,7 +1420,7 @@
   ASSERT_EQ(sink->mOnFrameCount, frame);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeMaxFsCodecChange) {
@@ -1437,7 +1437,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   unsigned int frame = 0;
   mVideoConduit->StartTransmitting();
@@ -1461,7 +1461,7 @@
   ASSERT_EQ(sink->mOnFrameCount, frame);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeMaxFsSinkWantsThenCodecChange) {
@@ -1478,7 +1478,7 @@
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
   wants.max_pixel_count = 3500 * 16 * 16;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   unsigned int frame = 0;
   mVideoConduit->StartTransmitting();
@@ -1501,7 +1501,7 @@
   ASSERT_EQ(sink->mOnFrameCount, frame);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeMaxFsNegotiated) {
@@ -1517,7 +1517,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   unsigned int frame = 0;
   mVideoConduit->StartTransmitting();
@@ -1550,7 +1550,7 @@
   ASSERT_EQ(sink->mOnFrameCount, frame);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 // Disabled: See Bug 1420493
@@ -1569,7 +1569,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(1280, 720, 1);
@@ -1591,7 +1591,7 @@
   ASSERT_EQ(sink->mOnFrameCount, 3U);
 
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeScaleResolutionBy) {
@@ -1609,7 +1609,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(1280, 720, 1);
@@ -1624,7 +1624,7 @@
   ASSERT_EQ(sink->mVideoFrame.timestamp_us(), 2000U);
   ASSERT_EQ(sink->mOnFrameCount, 2U);
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 TEST_F(VideoConduitTest, TestVideoEncodeSimulcastScaleResolutionBy) {
@@ -1649,7 +1649,7 @@
 
   UniquePtr<MockVideoSink> sink(new MockVideoSink());
   rtc::VideoSinkWants wants;
-  mVideoConduit->AddOrUpdateSink(sink.get(), wants);
+  mVideoConduit->AddOrUpdateSinkNotLocked(sink.get(), wants);
 
   mVideoConduit->StartTransmitting();
   SendVideoFrame(640, 480, 1);
@@ -1665,7 +1665,7 @@
   ASSERT_EQ(sink->mVideoFrame.timestamp_us(), 2000U);
   ASSERT_EQ(sink->mOnFrameCount, 2U);
   mVideoConduit->StopTransmitting();
-  mVideoConduit->RemoveSink(sink.get());
+  mVideoConduit->RemoveSinkNotLocked(sink.get());
 }
 
 }  // End namespace test.